core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -788,6 +788,20 @@ abstract class RDD[T: ClassTag](
   }
 
   /**
+   * Applies a function to each partition of this RDD, while tracking the index
+   * of the original partition.
+   */
+  def foreachPartitionWithIndex(
```

Comment:
Are you saying that you would prefer for users of this class to have to use mapPartitionsWithIndex with a side-effect of storing data, and then add an empty foreach in order to trigger the job to actually be scheduled?